packages/explorer/src/app/interact/Form.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import { Coins, Eye, Send } from "lucide-react";
+import { Abi, AbiFunction } from "viem";
+import { useDeferredValue, useState } from "react";
+import { Input } from "../../components/ui/Input";
+import { useHashState } from "../../hooks/useHash";
+import { cn } from "../../lib/utils";
+import { FunctionField } from "./FunctionField";
+
+type Props = {
+  data: {
+    abi: Abi;
+  };
+};
+
+export function Form({ data }: Props) {
+  const [hash] = useHashState();
+  const [filterValue, setFilterValue] = useState("");
+  const deferredFilterValue = useDeferredValue(filterValue);
+  const abiFunctions = data.abi.filter(
+    (abi) => (abi as AbiFunction).type === "function",
+  ) as AbiFunction[];
+  const filteredFunctions = abiFunctions.filter((abi) => {
+    return (abi as AbiFunction).name
+      .toLowerCase()
+      .includes(deferredFilterValue.toLowerCase());
+  });
+
+  return (
+    <div className="flex">
+      <div className="w-[350px]">
+        <div className="sticky top-16 pr-4">
+          <h4 className="py-4 text-xs font-semibold uppercase opacity-70">
+            Jump to:
+          </h4>
+
+          <Input
+            type="text"
+            placeholder="Filter functions..."
+            value={deferredFilterValue}
+            onChange={(evt) => {
+              setFilterValue(evt.target.value);
+            }}
+          />
+
+          <ul
+            className="mt-4 max-h-max space-y-2 overflow-y-auto pb-4"
+            style={{
+              maxHeight: "calc(100vh - 160px)",
+            }}
+          >
+            {filteredFunctions.map((abi, idx) => {
+              if ((abi as AbiFunction).type !== "function") {
+                return null;
+              }
+
+              return (
+                <li key={idx}>
+                  <a
+                    href={`#${(abi as AbiFunction).name}`}
```

Comment:
why does this type need to be casted if we already have it casted above in the array definition?

Reply:
Good catch, that's a leftover